src/Adapter/ArrayAdapter.php

```diff
@@ -21,17 +28,33 @@ public function send($queue, $message)
         return $d['id'];
     }
 
-    public function deleteMessage($receipt, $queue)
+    public function receive($queue)
     {
+        if (!array_key_exists($queue, $this->data)) {
+            throw new OutOfBoundsException("Missing data");
+        }
+
+        $values = array_values($this->data[$queue]);
+
+        $message = [$values[0]["id"], $values[0]["content"]];
+        $this->data[$queue][] = ["id" => $values[0]["id"], "content" => $values[0]["content"]];
+        array_shift($this->data[$queue]);
+
+        return $message;
     }
 
-    public function receive($queue)
+    public function delete($receipt, $queue)
     {
-        if (!array_key_exists($queue, $this->data)) {
-            throw \Exception();
+        if (array_key_exists($queue, $this->data)) {
+            foreach ($this->data[$queue] as $index => $data) {
+                if ($this->data[$queue][$index]["id"] == $receipt) {
+                    unset($this->data[$queue][$index]);
+                    return true;
+                }
+            }
         }
-        $row = $this->data[$queue->getName()][0];
-        return [$row['id'], $row['content']];
+
+        throw new InvalidArgumentException("Missing message with receipt: {$receipt}");
```

Comment:
This could be safe because we are effectively trying to do something impossible...